[hrishikeshsathyian] iP

hrishikeshsathyian opened the pull request:
# Softess
> The sun don't shine forever, but as long as it's here, then we might as well shine together. - P Diddy 

Softess, your one stop task tracking companion 🔥 . It's,
- easy to learn 📖 
- remembers your data 
- visually clean 
- *intuitive*

All you need to do is, 

1. download it from [here](https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.istockphoto.com%2Fphotos%2Fbanana-eating-monkey&psig=AOvVaw1rOEbOaj0vKgiaRV9a5YGs&ust=1738912831046000&source=images&cd=vfe&opi=89978449&ved=0CBQQjRxqFwoTCMjc257BrosDFQAAAAAdAAAAABAE) 
2. double-click it
3. manage your commitments and responsibilities


And it is **COMPLETELY FREE!**

Features: 
- [ ] Managing tasks 
- [ ] Managing deadlines 
- [ ] Managing reminders

This is how you see your current tasks, in terminal type : `list`

If you are a Java programmer, this is how the `main` method looks like:
``` Java

    public static void main(String[] args) {
        new Softess(FILE_PATH).run();
    }

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

hrishikeshsathyian committed:
Rename, Greet, Exit

hrishikeshsathyian committed:
Echo

hrishikeshsathyian committed:
Add, List

hrishikeshsathyian committed:
Mark as Done

hrishikeshsathyian committed:
ToDos, Events, Deadlines

hrishikeshsathyian committed:
A-TextUiTesting

hrishikeshsathyian committed:
Handling Errors

hrishikeshsathyian committed:
Delete

hrishikeshsathyian committed:
Save

hrishikeshsathyian committed:
Merge branch 'branch-Level-7'

hrishikeshsathyian committed:
base level working

hrishikeshsathyian committed:
Bug fies

hrishikeshsathyian committed:
Merge branch branch-Level-8

hrishikeshsathyian committed:
OOP-ify

hrishikeshsathyian committed:
A-MoreOOP

hrishikeshsathyian committed:
Divide classes into packages

hrishikeshsathyian committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2425S2/ip

hrishikeshsathyian committed:
Set up Gradle

hrishikeshsathyian committed:
Some JUnit Tests

hrishikeshsathyian committed:
Add JavaDoc Headers

hrishikeshsathyian committed:
Some Coding Standard changes

hrishikeshsathyian committed:
Add Find Functionality

hrishikeshsathyian committed:
A-JavaDoc

hrishikeshsathyian committed:
Add JavaDoc

hrishikeshsathyian committed:
Find Functionality

lihongong commented on src/main/java/softess/Deadline.java:
```diff
@@ -0,0 +1,89 @@
+public class Deadline extends Task {
+
+    /** The deadline date as a string */
+    protected String by;
```
Could name it as something more understandable like `deadline`.

lihongong commented on src/main/java/softess/Event.java:
```diff
@@ -0,0 +1,78 @@
+public class Event extends Task {
+
+    /** The starting time of the event */
+    protected String from;
+
+    /** The ending time of the event */
+    protected String to;
```
Could name it as something more understandable like `start` and `end`.

lihongong commented on src/main/java/softess/Task.java:
```diff
@@ -0,0 +1,74 @@
+    /**
+     * Marks this task as not completed and prints a confirmation message.
+     */
+    public void markAsUnDone() {
```
`markAsUnDone` should follow camelCase as `markAsUndone`.
I noticed the same issue in several other places too.

lihongong commented on src/main/java/softess/UserInterface.java:
```diff
@@ -0,0 +1,41 @@
+    /**
+     * Displays a goodbye message when the user exits the application.
+     */
+    public void showGoodByeMessage() {
```
`showGoodByeMessage` should follow camelCase as `showGoodbyeMessage`.

lihongong commented on src/main/java/softess/DataHandler.java:
```diff
@@ -0,0 +1,91 @@
+                String[] taskData = s.nextLine().split("\\|");
+                boolean status = Objects.equals(taskData[1], "1");
+
+                switch (taskData[0]) {
+                    case "TODO":
+                        tasks.add(new ToDo(taskData[2], status));
+                        break;
+                    case "DEADLINE":
+                        tasks.add(new Deadline(taskData[2], taskData[3], status));
+                        break;
+                    case "EVENT":
+                        tasks.add(new Event(taskData[2], taskData[3], taskData[4], status));
+                        break;
+                }
```
`switch` statement should follow the correct coding standard.

lihongong commented on src/main/java/softess/Command.java:
```diff
@@ -0,0 +1,23 @@
+package softess;
+
+
+public abstract class Command {
```
Should there be a header comment for this class?
I noticed the same issue in several other places too.

lihongong reviewed (commented):
Overall, I found your code easy to read for the most part.
I noted a few minor coding standard violations too.
I suggest to put all the Command and Task classes in separate packages to organize the files.

hrishikeshsathyian committed:
Implement GUI

hrishikeshsathyian committed:
Add GUI

hrishikeshsathyian committed:
Change to push new branch

hrishikeshsathyian committed:
Revert "Change to push new branch"

This reverts commit de1496d8fea8363deac364cb33fb5e14ba41d587.

hrishikeshsathyian committed:
Add assertions

hrishikeshsathyian committed:
Merge pull request #2 from hrishikeshsathyian/branch-A-Assertions

Add assertions

hrishikeshsathyian committed:
Merge main into CodeQuality

hrishikeshsathyian committed:
Improve code quality

hrishikeshsathyian committed:
Merge pull request #3 from hrishikeshsathyian/branch-A-CodeQuality

Branch a code quality

hrishikeshsathyian committed:
Add reminders for tasks due in 3 days

hrishikeshsathyian committed:
Merge B-Reminders

hrishikeshsathyian committed:
Add additional JavaDoc comments

Writing additional JavaDoc comments helps improve code quality.

hrishikeshsathyian committed:
Add more personalised and informative error message

Displaying more informative error messages gives a better user experience.

Giving Softess more personality also makes it more enjoyable.

hrishikeshsathyian committed:
Add representative screenshot

hrishikeshsathyian committed:
Add ReadMe

hrishikeshsathyian committed:
Relocate data file

hrishikeshsathyian committed:
Fix data read/write bug

hrishikeshsathyian committed:
Update gradle file
